### ISSUE TITLE
NPM Plugin: Pass exact flag to lerna publish

### ISSUE THREAD

**Issue by bmuenzenmeyer** [intuit/auto#1214]
**Describe the bug**

The lerna docs state that the --exact flag is supported in both [publish](https://github.com/lerna/lerna/tree/master/commands/publish#options) and [version](https://github.com/lerna/lerna/tree/master/commands/version#--exact) commands. It seems, however, that they each handle the flag differently depending on the architecture of the monorepo, specifically around dependencies defined via `file://` specifiers, which our team is using.

❌ In `lerna version`, the [flag is ignored](https://github.com/lerna/lerna/blob/master/commands/version/index.js#L522-L525) when `file://` dependencies are found.
✅ In `lerna publish`, the [flag is honored](https://github.com/lerna/lerna/blob/master/commands/version/index.js#L522-L525)

If unfamiliar, a relative file specifier in a `package.json` looks like this:

```json
  "dependencies": {
    "@scope/foo": "file:../foo",
    "@scope/bar": "file:../bar",
    "chalk": "2.4.2"
  },
```

I learned that our existing tooling that directly calls `lerna version` and then `lerna publish` is doing thing same thing that `auto` is - only supplying the `--exact` flag to `lerna version`.

**To Reproduce**

This is a bit more involved. I can create a monorepo with this structure if absolutely needed, but to test it, you need to run auto and see the publications anyways... 🤷. I'm hoping that you'll both take my word on it, and in reading the above source code from lerna, see how the bug manifests.

I verified that passing `--exact` directly to `lerna publish` works, even with this architecture. 

**Expected behavior**

npm plugin `--exact` is supported for monorepos with `file://` dependencies

**Screenshots**

**Environment information:**

Environment Information:

"auto" version: v9.31.1
"git"  version: v2.20.1 (Apple Git-117)
"node" version: v12.14.1GHE version:    v2.19.4

Project Information:

✔ Repository:      REDACTED
✔ Author Name:     BrianMuenzenmeyer
✔ Author Email:    REDACTED
✖ Current Version: 
✔ Latest Release:  REDACTED

✔ Labels configured on GitHub project 

GitHub Token Information:

✔ Token:            REDACTED
✔ Repo Permission:  admin
✔ User:             BrianMuenzenmeyer
✔ API:              REDACTED
✔ Enabled Scopes:   read:user, repo
✔ Rate Limit:       29945/30000

**Additional context**

I'd be willing to contribute a fix for this, if accepted. I think the flag need only be conditionally applied [within the publish hook here](https://github.com/intuit/auto/blob/master/plugins/npm/src/index.ts#L921) as it is [here in the version hook](https://github.com/intuit/auto/blob/master/plugins/npm/src/index.ts#L641).

Alternatively, if we tapped into `auto.hooks.publish.tapPromise` - that wouldn't completely replace the implementation of the npm plugin, correct? It would just call both.... is my working understanding of how tap works.

**Comment by hipstersmoothie** [comment]
Nice catch! PR incoming. 

> Alternatively, if we tapped into auto.hooks.publish.tapPromise - that wouldn't completely replace the implementation of the npm plugin, correct? It would just call both.... is my working understanding of how tap works.

That particular hook is an `AsyncParallelHook` so you wouldn't be able to replace the behavior.

**Comment by adierkens** [comment]
:rocket: Issue was released in [`v9.32.2`](https://github.com/intuit/auto/releases/tag/v9.32.2) :rocket:
